ext/opentelemetry-ext-opentracing-shim/src/opentelemetry/ext/opentracing_shim/__init__.py

```diff
@@ -98,18 +284,39 @@ def get_baggage_item(self, key):
 
 class ScopeShim(opentracing.Scope):
     """A `ScopeShim` wraps the OpenTelemetry functionality related to span
-    activation/deactivation while using OpenTracing `Scope` objects for
-    presentation.
+    activation/deactivation while using OpenTracing :class:`opentracing.Scope`
+    objects for presentation.
+
+    Unlike other classes in this package, the `ScopeShim` class doesn't wrap an
+    OpenTelemetry class because OpenTelemetry doesn't have the notion of
+    "scope" (though it *does* have similar functionality).
 
     There are two ways to construct a `ScopeShim` object: using the default
-    initializer and using the `from_context_manager()` class method.
+    initializer and using the :meth:`from_context_manager()` class method.
 
     It is necessary to have both ways for constructing `ScopeShim` objects
-    because in some cases we need to create the object from a context manager,
-    in which case our only way of retrieving a `Span` object is by calling the
-    `__enter__()` method on the context manager, which makes the span active in
-    the OpenTelemetry tracer; whereas in other cases we need to accept a
-    `SpanShim` object and wrap it in a `ScopeShim`.
+    because in some cases we need to create the object from an OpenTelemetry
+    `Span` context manager (as returned by
+    :meth:`opentelemetry.trace.Tracer.use_span`), in which case our only way of
+    retrieving a `Span` object is by calling the ``__enter__()`` method on the
+    context manager, which makes the span active in the OpenTelemetry tracer;
+    whereas in other cases we need to accept a `SpanShim` object and wrap it in
+    a `ScopeShim`. The former is used mainly when the instrumentation code
+    retrieves the currently-active span using `ScopeManagerShim.active`. The
+    latter is mainly used when the instrumentation code activates a span using
+    :meth:`ScopeManagerShim.activate`.
+
+    Args:
+        manager: The :class:`ScopeManagerShim` that created this
+            :class:`ScopeShim`.
+        span: The :class:`SpanShim` this :class:`ScopeShim` controls.
+        span_cm(:class:`contextlib.AbstractContextManager`, optional): A
+            Python context manager which yields an OpenTelemetry `Span` from
+            its ``__enter__()`` method. Used by :meth:`from_context_manager` to
+            store the context manager as an attribute so that it can later be
+            closed by calling its ``__exit__()`` method. Defaults to `None`.
+
+    TODO: Is :class:`contextlib.AbstractContextManager` the correct type for *span_cm*?
```

Comment:
Maybe `typing.ContextManager`.

Reply:
I recall trying this and getting shouted at by Mypy while trying to add typing to the shim (don't have the error at the moment).
Anyway, I guess this is relevant only if we add typing to the shim, right?

Reply:
Yeah, I think you can use `ContextManager` and remove the TODO.